e3db/types/note.py

```diff
@@ -0,0 +1,356 @@
+            additional functionality. 
+
+        Returns
+        -------
+        None
+
+        """
+        self.__client_id                = note_options.__client_id
+        self.__mode                     = note_keys.__mode
+        self.__recipient_signing_key    = note_keys.__recipient_signing_key
+        self.__writer_signing_key       = note_keys.__writer_signing_key
+        self.__writer_encryption_key    = note_keys.__writer_encryption_key
+        self.__encrypted_access_key     = note_keys.__encrypted_access_key
+        self.__type                     = note_options.__type
+        self.__data                     = copy.deepcopy(data)
+        self.__plain                    = note_options.__plain
+        self.__file_meta                = note_options.__file_meta
+        self.__max_views                = note_options.__max_views
+        self.__expiration               = note_options.__expiration
+        self.__expires                  = note_options.__expires
+        self.__eacp                     = None 
+
+        self.__note_id                  = None
+        self.__created_at               = None # Is this server defined as per the JS implementation's comment? (see note.js)
+
+        # Signing makes sense to do after Note creation from a user's perspective.
+        # Makes less sense to, say, "create this note with a signature you already have"
+        # rather than "create a note, then sign it when you want to write it". 
+        self.__signature                = "" 
+
+    def to_json(self):
+        """
+        Seriaizes the note as a JSON-style object.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        dict
+            JSON-style document containing the Note elements
+        """
+        to_serialize = {
+            'client_id': self.__client_id,
+            'mode': self.__mode,
+            'recipient_signing_key': self.__recipient_signing_key,
+            'writer_signing_key': self.__writer_signing_key,
+            'writer_encryption_key': self.__writer_encryption_key,
+            'encrypted_access_key': self.__encrypted_access_key,
+            'type': self.__type,
+            'data': self.__data,
+            'plain': self.__plain,
+            'file_meta': self.__file_meta,
+            'signature': self.__signature,
+            'max_views': self.__max_views,
+            'expiration': self.__expiration,
+            'expires': self.__expires,
+            'eacp': self.__eacp,
+            'created_at': self.__created_at,
+            'note_id': self.__note_id
+        }
+
+        # Removes any None values. This is done in Meta.to_json(), unsure if necessary 
+        # here. 
+        for key, value in list(to_serialize.items()):
+            if value is None or value == 'None':
+                del to_serialize[key]
+
+        to_serialize
+
+    # Which fields should have getters and setters? Do notes need setters as a record does? 
```

Comment:
For a language like python I’d argue only computed fields (like signature) need getters and setters, and `self.__data` should be `self.data` . To me we’re only adding busy work overhead on adding a getter or setter that does what a user can do in the same number of code lines (set or get a class value).